pkg/auth/dependency/interaction/deps.go

```diff
@@ -52,12 +54,17 @@ func ProvideUserProvider(
 		Hooks:                         hp,
 		URLPrefix:                     up,
 		TaskQueue:                     q,
-		WelcomeEmailConfiguration:     config.AppConfig.WelcomeEmail,
 		UserVerificationConfiguration: config.AppConfig.UserVerification,
+		WelcomeMessageProvider:        wmp,
 	}
 }
 
+func ProvideWelcomeMessageProvider(p *welcomemessage.Provider) WelcomeMessageProvider {
```

Comment:
For cross-package dependencies (e.g. `interaction` -> `welcomemessage`), it's better to define it using `wire.Bind` at the composition root to avoid circular dependency and reduce coupling.